pkg/utils/workloadidentity/jwt.go

```diff
@@ -0,0 +1,191 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package workloadidentity
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/go-jose/go-jose/v4"
+	"github.com/go-jose/go-jose/v4/jwt"
+)
+
+const (
+	keyUsageSig string = "sig"
+)
+
+var (
+	now = time.Now
+)
+
+// TokenIssuer is JSON Web Token issuer.
+type TokenIssuer struct {
+	signer             jose.Signer
+	issuer             string
+	minDurationSeconds int64
+	maxDurationSeconds int64
+}
+
+// NewTokenIssuer creates new TokenIssuer.
+func NewTokenIssuer(signingKey any, issuer string, minDuration, maxDuration int64) (*TokenIssuer, error) {
+	signer, err := getSigner(signingKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get signer: %w", err)
+	}
+
+	return &TokenIssuer{
+		signer:             signer,
+		issuer:             issuer,
+		minDurationSeconds: minDuration,
+		maxDurationSeconds: maxDuration,
+	}, nil
+}
+
+// GetIssuer returns the issuer value used for the `iss` JWT claim.
+func (t *TokenIssuer) GetIssuer() string {
```

Comment:
Nit: I would call this `Issuer` to align with https://go.dev/doc/effective_go#Getters

```suggestion
// Issuer returns the issuer value used for the `iss` JWT claim.
func (t *TokenIssuer) Issuer() string {
```